src/web/jcc/client/components/SetHubLocationPanel.tsx

```diff
@@ -0,0 +1,189 @@
+import { getElementById, getGeographicCoordinate } from './shared/Utilities';
+import { Map } from 'ol';
+import PointerInteraction from 'ol/interaction/Pointer';
+import { Select, MenuItem } from '@mui/material';
+
+interface Props {
+    hubs: {[key: number]: PortalHubStatus}
+    api: JaiaAPI
+    map: Map
+}
+
+
+export default function SetHubLocationPanel(props: Props) {
+    const [ hub_id, set_hub_id ] = useState(Number(Object.keys(props.hubs)[0]) ?? 1)
+    const [ selectingOnMap, setSelectingOnMap ] = useState(false)
+    const selectOnMapInteractionRef = useRef(null)
+
+    const hubLocation = props.hubs[hub_id].location
+
+    /**
+     * A Select element for choosing the hub_id.
+     *
+     * @returns {*} The Select element.
+     */
+    function hubIdSelectionElement() {
+        const menuItems = Object.values(props.hubs).map((hub) => {
+            return <MenuItem value={hub.hub_id}>{`Hub ${hub.hub_id}`}</MenuItem>
+        })
+
+        const labelRow = <div className='mission-settings-input-label'>
+                Hub ID
+            </div>
+
+        const hubIdSelectionRow = 
+            <div className='mission-settings-input-row'>
+                <Select id="bot-id-select" defaultValue={hub_id} onChange={(evt) => {
+                        set_hub_id(Number(evt.target.value))
+                    }}>
+                    { menuItems }
+                </Select>
+            </div>
+
+        return [ labelRow, hubIdSelectionRow ]
+    }
+
+
+    
+    /**
+     * Gets the hub location from the text inputs.
+     *
+     * @returns {GeographicCoordinate}
+     */
+    function getInputHubLocation(): GeographicCoordinate {
+        const lat = Number(getElementById<HTMLInputElement>('set-hub-location-latitude').value)
+        const lon = Number(getElementById<HTMLInputElement>('set-hub-location-longitude').value)
+        if (lat == null || lon == null) {
+            return null
+        }
+
+        return {
+            lat: lat,
+            lon: lon
+        }
+    }
+
+
+    /**
+     * Calls the API to submit a location change for a certain hub.
+     *
+     */
+    function submitHubLocation() {
+        const hubLocation = getInputHubLocation()
+
+        if (hubLocation == null) {
+            console.warn('hub location is null')
+            return
+        }
+
+        const hubCommand: CommandForHub = {
+            hub_id: hub_id,
+            type: HubCommandType.SET_HUB_LOCATION,
+            set_hub_location: hubLocation
+        }
+
+        props.api.postCommandForHub(hubCommand)
+    }
+
+    
+    /**
+     * Initiate a PointerInteraction to select a new hub location with a click or tap.
+     */
+    const toggleSelectOnMapInteraction = () => {
+        if (selectOnMapInteractionRef.current != null) {
+            destroySelectOnMapInteraction()
+            return
+        }
+
+        selectOnMapInteractionRef.current = new PointerInteraction({
+            handleEvent: (evt) => {
```

Comment:
From the OpenLayers docs:

```Method called by the map to notify the interaction that a browser event was dispatched to the map. The function may return false to prevent the propagation of the event to other interactions in the map's interactions chain.```

So, I'm preventing the click event from falling through to the other interactions on the map's interaction stack.  I added some comments to clarify the reasoning on this.